src/sparseml/pytorch/utils/exporter.py

```diff
@@ -570,7 +570,23 @@ def export_onnx(
 
         # clean up graph from any injected / wrapped operations
         _delete_trivial_onnx_adds(onnx_model)
-    onnx.save(onnx_model, file_path)
+
+    try:
+        # alternatively we can just check the size of the onnx file on disk
```

Comment:
To decide whether we should save the model with/without external data we could either check its disk size (feels over-engineered) or simply check whether we fail on saving all the data directly to protobuf. If yes, we switch to exporting with the external data file.